src/libraries/System.Text.Json/tests/Serialization/ContinuationTests.cs

```diff
@@ -10,502 +11,314 @@
-                Assert.Null(obj.C.F);
-                Assert.True(obj.C.G);
-                Assert.Null(obj.C.H);
-                Assert.Collection(obj.C.I, v => Assert.Equal(42, v), v => Assert.Equal(17, v));
-                Assert.NotNull(obj.C.J);
-                Assert.Null(obj.C.J.A);
-                Assert.Equal(7, obj.C.J.B);
-            }
-        }
+                        // Determine the DefaultBufferSize that is required to contain the complete json.
+                        int bufferSize = 16;
+                        while (payload.Length > bufferSize)
+                        {
+                            bufferSize *= 2;
+                        }
+                        int minPaddingLength = bufferSize - payload.Length + 1;
+                        int maxPaddingLength = bufferSize - 1;
 
-        [Theory]
-        [MemberData(nameof(TestData))]
-        public static async Task ContinuationShouldWorkAtAnyPosition_Class_ValueType(int paddingLength, bool ignoreNullValues)
-        {
-            var stream = new MemoryStream();
-            {
-                var obj = new Outer<TestClass<NestedValueType>>
-                {
-                    S = new string('x', paddingLength),
-                    C = new()
-                    {
-                        A = "Hello",
-                        B = null,
-                        C = 42,
-                        D = null,
-                        E = 3.14e+17f,
-                        F = null,
-                        G = true,
-                        H = null,
-                        I = new int[] { 42, 17 },
-                        J = new()
+                        foreach (int length in Enumerable.Range(minPaddingLength, maxPaddingLength - minPaddingLength + 1))
                         {
-                            A = null,
-                            B = 7,
+                            foreach (bool ignore in IgnoreNullValues)
+                            {
+                                yield return new object[] { new string(' ', length) + payload, bufferSize, arrayType, ignore };
+                            }
                         }
                     }
-                };
-                await JsonSerializer.SerializeAsync(stream, obj, new JsonSerializerOptions { Converters = { new OuterConverter<TestClass<NestedClass>>() } });
+                }
             }
+        }
 
-            stream.Position = 0;
+        [Theory]
+        [MemberData(nameof(TestData), /* enumeratePayloadTweaks: */ false)]
+        public static async Task ShouldWorkAtAnyPosition_Stream(string json, int bufferSize, Type type, bool ignoreNullValues)
+        {
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(json));
             {
                 var readOptions = new JsonSerializerOptions
                 {
-                    DefaultBufferSize = 128,
+                    DefaultBufferSize = bufferSize,
```

Comment:
Can we also add a permutation where `DefaultBufferSize` is not changed?

Reply:
Setting the buffer size to the calculated value is crucial for these tests. That is how the continuation point is determined. Using the default value here leads to actual no continuation at all.